src/components/expandable_section.rs

```diff
@@ -129,21 +132,47 @@ pub fn expandable_section(props: &ExpandableSectionProperties) -> Html {
         class.extend(classes!("pf-m-expanded"));
     }
 
+    let content = html!(
+        <div
+            class="pf-v5-c-expandable-section__content" hidden={!expanded}
+        >{ props.children.clone() }</div>
+    );
+
+    let truncating = matches!(props.variant, ExpandableSectionVariant::Truncate);
+    let toggle = match props.detached {
+        true => html!(),
+        false => html!(
+            <ExpandableSectionToggle
+                {ontoggle}
+                expanded={expanded}
+                toggle_text_hidden={&props.toggle_text_hidden}
+                toggle_text_expanded={&props.toggle_text_expanded}
+                detached=false
+                direction={ExpandableSectionToggleDirection::Down}
+                no_icon={truncating}
+            />
+        ),
+    };
+
+    let content = match truncating {
+        false => html!(
+            <>
+                {toggle} {content}
+            </>
+        ),
+        true => html!(
+            <>
+                 {content} {toggle}
+            </>
+        ),
+    };
+
     html!(
-        <div {class}>
-            if !props.detached {
-                <ExpandableSectionToggle
-                    {ontoggle}
-                    expanded={expanded}
-                    toggle_text_hidden={&props.toggle_text_hidden}
-                    toggle_text_expanded={&props.toggle_text_expanded}
-                    detached=false
-                    direction={ExpandableSectionToggleDirection::Down}
-                />
-            }
-          <div
-                class="pf-v5-c-expandable-section__content" hidden={!expanded}
-          >{ props.children.clone() }</div>
+        <div
+            {class}
```

Comment:
I think you dropped the `hidden` attribute

Reply:
No, that's still part of the `content` variable. But move a bit more to the top of the function.